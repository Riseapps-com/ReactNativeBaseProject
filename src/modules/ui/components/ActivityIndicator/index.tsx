import React from 'react';
import {
  ActivityIndicator as NativeActivityIndicator,
  ActivityIndicatorProps as NativeActivityIndicatorProps,
  View,
} from 'react-native';

import { useTheme } from '~theme';

import themedStyles from './styles';

export type ActivityIndicatorProps = NativeActivityIndicatorProps;

const ActivityIndicator: React.FC<ActivityIndicatorProps> = props => {
  const [styles, theme] = useTheme(themedStyles);

  return (
    <View style={styles.wrapper}>
      <NativeActivityIndicator color={theme.primary} size={'large'} {...props} />
    </View>
  );
};

export default ActivityIndicator;
