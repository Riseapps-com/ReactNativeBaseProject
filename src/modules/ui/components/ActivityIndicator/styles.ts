import { styleSheetFactory } from '~theme';

export default styleSheetFactory(() => ({
  wrapper: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
}));
