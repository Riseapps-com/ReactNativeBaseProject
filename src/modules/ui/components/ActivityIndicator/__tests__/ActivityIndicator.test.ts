import { renderComponent } from '~modules/tests';

import ActivityIndicator from '../index';

const renderActivityIndicator = () => renderComponent(ActivityIndicator);

describe('ui', () => {
  describe('<ActivityIndicator />', () => {
    it('renders <ActivityIndicator />', () => {
      const activityIndicator = renderActivityIndicator();

      expect(activityIndicator).toMatchSnapshot();
    });
  });
});
