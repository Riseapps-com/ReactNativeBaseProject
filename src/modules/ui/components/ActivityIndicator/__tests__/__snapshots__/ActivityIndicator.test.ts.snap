// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui <ActivityIndicator /> renders <ActivityIndicator /> 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    accessibilityLabel="Activity indicator"
    color="#0036A7"
    size="large"
  />
</View>
`;
